lib/Sema/TypeCheckEffects.cpp

```diff
@@ -3581,27 +3583,43 @@ class CheckEffectsCoverage : public EffectsHandlingWalker<CheckEffectsCoverage>
     Ctx.Diags.diagnose(E->getAwaitLoc(), diag::no_async_in_await);
   }
 
-  void diagnoseUncoveredAsyncSite(const Expr *anchor) const {
-    auto asyncPointIter = uncoveredAsync.find(anchor);
-    if (asyncPointIter == uncoveredAsync.end())
-      return;
-    const std::vector<DiagnosticInfo> &errors = asyncPointIter->getSecond();
+  std::pair<SourceLoc, std::string>
+  getFixItForUncoveredAsyncSite(const Expr *anchor) const {
     SourceLoc awaitInsertLoc = anchor->getStartLoc();
-    if (const AnyTryExpr *tryExpr = dyn_cast<AnyTryExpr>(anchor))
+    std::string insertion = "await ";
+    if (auto *tryExpr = dyn_cast<AnyTryExpr>(anchor))
       awaitInsertLoc = tryExpr->getSubExpr()->getStartLoc();
-    else if (const AutoClosureExpr *autoClosure = dyn_cast<AutoClosureExpr>(anchor)) {
-      if (const AnyTryExpr *tryExpr = dyn_cast<AnyTryExpr>(autoClosure->getSingleExpressionBody()))
+    else if (auto *autoClosure = dyn_cast<AutoClosureExpr>(anchor)) {
+      if (auto *tryExpr = dyn_cast<AnyTryExpr>
+          (autoClosure->getSingleExpressionBody()))
         awaitInsertLoc = tryExpr->getSubExpr()->getStartLoc();
+      // Supply a tailored fixIt including the identifier if we are
+      // looking at a shorthand optional binding.
+    } else if (anchor->isImplicit()) {
+      if (auto declRef = dyn_cast<DeclRefExpr>(anchor))
+        if (auto var = dyn_cast_or_null<VarDecl>(declRef->getDecl())) {
+          insertion = " = await " + var->getNameStr().str();
+          awaitInsertLoc = Lexer
+          ::getLocForEndOfToken(Ctx.Diags.SourceMgr, anchor->getStartLoc());
+        }
     }
+    return std::make_pair(awaitInsertLoc, insertion);
+  }
 
+  void diagnoseUncoveredAsyncSite(const Expr *anchor) const {
+    auto asyncPointIter = uncoveredAsync.find(anchor);
+    if (asyncPointIter == uncoveredAsync.end())
+      return;
+    const std::vector<DiagnosticInfo> &errors = asyncPointIter->getSecond();
+    const auto fixIt = getFixItForUncoveredAsyncSite(anchor);
```

Comment:
Nit: would this syntax work
```suggestion
    const auto &[loc, insertText] = getFixItForUncoveredAsyncSite(anchor);
```

Reply:
I will try it thanks!

Reply:
It works, much cleaner, thanks for showing.